src/components/demo-step-0.js

```diff
@@ -0,0 +1,107 @@
+import React, { Component } from 'react'
+import { render } from 'react-dom'
+
+import contractService from '../services/contract-service'
+import ipfsService from '../services/ipfs-service'
+import bs58 from 'bs58'
+
+class DemoStep0 extends Component {
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      listingsResults: [],
+      contractListingsCount: -1
+    }
+
+    let that = this
+
+    // Test getting listings from chain
+    setTimeout(function() {
+      // TODO: Remove hacky 2s delay and correctly determine when contractService
+      // is ready
+      contractService.getAllListings().then((allContractResults) => {
+        var resultIndex;
```

Comment:
Should this be a let?

Reply:
YES!

Reply:
..in my defense, realized they actually use `var` in the official docs.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in
I'm changing it to `.forEach()` which, also solves the problem of keeping reference to the result object. Cleaner code all around!
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach